File: pipeline/analyze/parse_mappings.py
#!/usr/bin/env python3
"""
Parse region/appellation mappings from Claude tool output JSON.

Usage:
    python -m pipeline.analyze.parse_mappings <json-file>
"""

import json
import re
import sys
from pathlib import Path

sys.path.insert(0, str(Path(__file__).resolve().parents[2]))


def main():
    if len(sys.argv) < 2:
        print("Usage: python -m pipeline.analyze.parse_mappings <json-file>")
        sys.exit(1)

    raw = Path(sys.argv[1]).read_text(encoding="utf-8")
    parsed = json.loads(raw)
    text = parsed[0]["text"]
    match = re.search(r"\[\{.*\}\]", text, re.DOTALL)
    if not match:
        print("No JSON array found in text")
        sys.exit(1)

    data = json.loads(match.group(0))
    print(f"Total mappings with NULL appellation: {len(data)}")
    print("\n--- All entries by candidate_count ---")
    for i, d in enumerate(data):
        idx = str(i + 1).rjust(3)
        rn = (d.get("region_name") or "").ljust(45)
        co = (d.get("country") or "").ljust(18)
        cnt = str(d.get("candidate_count", 0)).rjust(5)
        res = d.get("resolved_region") or "NULL"
        ca = "catch-all" if d.get("is_catch_all") else "real"
        mt = d.get("match_type") or ""
        print(f"{idx}. {rn} | {co} | {cnt} wines | -> {res} ({ca}) | {mt}")


if __name__ == "__main__":
    main()

File: scripts_archive/node/root/parse_mappings.mjs
import { readFileSync } from 'fs';
const raw = readFileSync('C:\\Users\\neilw\\.claude\\projects\\C--Users-neilw-Documents-GitHub-loamv2\\2fc0b9c8-321b-4022-baf5-1ae0f9958732\\tool-results\\toolu_01JzqNDFX3w6LmAtEh1bJtVw.json', 'utf-8');
const parsed = JSON.parse(raw);
const text = parsed[0].text;
const match = text.match(/\[\{.*\}\]/s);
if (match) {
  const data = JSON.parse(match[0]);
  console.log('Total mappings with NULL appellation:', data.length);
  console.log('\n--- All entries by candidate_count ---');
  for (let i = 0; i < data.length; i++) {
    const d = data[i];
    const idx = String(i + 1).padStart(3);
    const rn = (d.region_name || '').padEnd(45);
    const co = (d.country || '').padEnd(18);
    const cnt = String(d.candidate_count).padStart(5);
    const res = d.resolved_region || 'NULL';
    const ca = d.is_catch_all ? 'catch-all' : 'real';
    const mt = d.match_type || '';
    console.log(`${idx}. ${rn} | ${co} | ${cnt} wines | → ${res} (${ca}) | ${mt}`);
  }
}
